src/waveform/renderers/allshader/waveformrendererrgb.cpp

```diff
@@ -185,13 +202,35 @@ void WaveformRendererRGB::paintGL() {
             blue *= normFactor;
         }
 
+        float red_ghost = ghostLow * low_r + ghostMid * mid_r + ghostHigh * high_r;
+        float green_ghost = ghostLow * low_g + ghostMid * mid_g + ghostHigh * high_g;
+        float blue_ghost = ghostLow * low_b + ghostMid * mid_b + ghostHigh * high_b;
+        const float maxComponent_ghost = math_max3(red_ghost, green_ghost, blue_ghost);
+        if (maxComponent_ghost == 0.f) {
+            // Avoid division by 0
+            red_ghost = 0.f;
+            green_ghost = 0.f;
+            blue_ghost = 0.f;
+        } else {
+            const float normFactor_ghost = 1.f / maxComponent_ghost;
+            red_ghost *= normFactor_ghost;
+            green_ghost *= normFactor_ghost;
+            blue_ghost *= normFactor_ghost;
+        }
+
         // Lines are thin rectangles
         m_vertices.addRectangle(fpos - 0.5f,
                 halfBreadth - heightFactor * maxAllChn[0],
                 fpos + 0.5f,
                 halfBreadth + heightFactor * maxAllChn[1]);
         m_colors.addForRectangle(red, green, blue);
 
+        m_vertices_ghost.addRectangle(fpos - 0.5f,
+                halfBreadth - heightFactor * maxAllChn_ghost[0],
+                fpos + 0.5f,
+                halfBreadth + heightFactor * maxAllChn_ghost[1]);
+        m_colors_ghost.addForRectangle(red_ghost, green_ghost, blue_ghost, ghost_alpha);
+
```

Comment:
Since the alpha doesn't change it would be better to use a uniform for that. It would be a simple to add the uniform for the alpha to the RGBShader, and that way you could use the same shader for normal and ghost.